File: scratchfiles/averagePair.js
function averagePair(arr, average) {
    // add whatever parameters you deem necessary - good luck!
    let obj = {}
    arr.forEach((n) => obj[n] = ((average * 2) - n))
    for (let key in obj) {
        let val = obj[key]
        if (obj[val]) {
            return true
        }
    }
    return false
}

console.log(averagePair([1, 2, 3], 2.5)) // true
averagePair([1, 3, 3, 5, 6, 7, 10, 12, 19], 8) // true
averagePair([-1, 0, 3, 4, 5, 6], 4.1) // false
averagePair([], 4) // false

File: scratchfiles/average_pair.rb
def average_pair(arr, average)
  paired_hash = arr.each_with_object({}) {|int, result| result[int.to_f] = average * 2.0 - int}
  paired_hash.each do |_k,v|
    if paired_hash[v]
      return true 
    end
  end
false 
end

p average_pair([1,2,3], 2.5)
p average_pair([1, 3, 3, 5, 6, 7, 10, 12, 19], 8) 
p average_pair([-1, 0, 3, 4, 5, 6], 4.1)